postgresql/resource_postgresql_physical_replication_slot.go

```diff
@@ -0,0 +1,135 @@
+package postgresql
+
+import (
+	"database/sql"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourcePostgreSQLPhysicalReplicationSlot() *schema.Resource {
+	return &schema.Resource{
+		Create: PGResourceFunc(resourcePostgreSQLPhysicalReplicationSlotCreate),
+		Read:   PGResourceFunc(resourcePostgreSQLPhysicalReplicationSlotRead),
+		Delete: PGResourceFunc(resourcePostgreSQLPhysicalReplicationSlotDelete),
+		Exists: PGResourceExistsFunc(resourcePostgreSQLPhysicalReplicationSlotExists),
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourcePostgreSQLPhysicalReplicationSlotCreate(db *DBConnection, d *schema.ResourceData) error {
+
+	name := d.Get("name").(string)
+	txn, err := startTransaction(db.client, "")
+	if err != nil {
+		return err
+	}
+	defer deferredRollback(txn)
+
+	sql := "SELECT FROM pg_create_physical_replication_slot($1)"
+	if _, err := txn.Exec(sql, name); err != nil {
+		return err
+	}
+
+	if err = txn.Commit(); err != nil {
+		return fmt.Errorf("error creating physical ReplicationSlot: %w", err)
+	}
+
+	d.SetId(name)
+
+	return resourcePostgreSQLPhysicalReplicationSlotReadImpl(db, d)
+}
+
+func resourcePostgreSQLPhysicalReplicationSlotExists(db *DBConnection, d *schema.ResourceData) (bool, error) {
+
+	var ReplicationSlotName string
+
+	replicationSlotName, err := getReplicationSlotName(d, db.client)
+	if err != nil {
+		return false, err
+	}
+
+	txn, err := startTransaction(db.client, "")
+	if err != nil {
+		return false, err
+	}
+	defer deferredRollback(txn)
+
+	query := "SELECT slot_name FROM pg_catalog.pg_replication_slots WHERE slot_name = $1 and slot_type = 'physical'"
+	err = txn.QueryRow(query, replicationSlotName).Scan(&ReplicationSlotName)
+	switch {
+	case err == sql.ErrNoRows:
+		return false, nil
+	case err != nil:
+		return false, err
+	}
+
+	return true, nil
+}
+
+func resourcePostgreSQLPhysicalReplicationSlotRead(db *DBConnection, d *schema.ResourceData) error {
+	return resourcePostgreSQLPhysicalReplicationSlotReadImpl(db, d)
```

Comment:
```suggestion
	d.Set("name") = d.Id()
```

We only need to fill the name in case of `terraform import` (based on the resource ID which is the name)